cypress/support/views.ts

```diff
@@ -65,10 +65,7 @@ declare global {
 }
 
 Cypress.Commands.add("viewRoomByName", (name: string): Chainable<JQuery<HTMLElement>> => {
-    return cy
-        .findByRole("treeitem", { name: new RegExp("^" + name) })
-        .should("have.class", "mx_RoomTile")
-        .click();
+    return cy.findByRole("treeitem", { name }).should("have.class", "mx_RoomTile").click();
```

Comment:
... Did you look at why its like this? Its described in the JSDoc.

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/32d068f1-9c63-4da4-a3c2-f8c8f51d1ace)

Reply:
Ah, I didn't spot that thanks. I'll take another look. I think we should check for it actually containing the exact text - this approach doesn't work for rooms with similar names.

Reply:
Another attempt in [a814f1d](https://github.com/matrix-org/matrix-react-sdk/pull/11440/commits/a814f1d66a0eafcc1d25b7c7368adcc000c9e986) - this finds the "Rooms" tree and then uses `findByText` inside there, which should find only the exact room name.